ninja-core/src/main/java/ninja/standalone/AbstractStandalone.java

```diff
@@ -0,0 +1,402 @@
+/**
+ * Copyright (C) 2012-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ninja.standalone;
+
+import ninja.utils.OverlayedNinjaProperties;
+import com.google.common.base.Optional;
+import com.google.inject.CreationException;
+import java.util.ArrayList;
+import java.util.List;
+import static ninja.standalone.StandaloneHelper.checkContextPath;
+import ninja.utils.NinjaConstant;
+import ninja.utils.NinjaMode;
+import ninja.utils.NinjaModeHelper;
+import ninja.utils.NinjaPropertiesImpl;
+import org.apache.commons.lang3.StringUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Abstract Standalone that implements most functionality required to write
+ * a concrete Standalone.  Introduces new doStart(), doStop(), and doJoin()
+ * methods which are actually where you'll place most of your logic.  You'll
+ * also want to subclass the configure() method to add any configuration
+ * specific to your Standalone.  See NinjaJetty for example concrete implementation.
+ */
+abstract public class AbstractStandalone<T extends AbstractStandalone> implements Standalone<T>, Runnable {
```

Comment:
I don't want to start a big discussion - usually composition worked better for me in the long run... But it's okay for me the way it is now...